### PR TITLE
[WIP] Add easter eggs to Liberty

### PR DESCRIPTION
**Pull request by Earth-Cake**
Adding easter eggs to Liberty and fixing a bug.

### CODE REVIEW

**Comment by Earth-Cake** [comment]
@ln-zookeeper can you give your thoughts?

File: data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg [hunk 1]
```diff
@@ -606,6 +617,205 @@
...
+                [message]
+                    speaker=Ynyr
+                    message= _ "I'll meet you in the underground."
+                [/message]
+            [/then]
+
+            [else]
+                [message]
+                    speaker=Ynyr
+                    message= _ "Honey, I am going to meet our Gods."
```
Comment (Wedge009) [inline data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg]: Advise lower-case 'gods'.

File: data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg [hunk 2]
```diff
@@ -606,6 +617,205 @@
...
+                    message= _ "I am so sorry..."
+                [/message]
+            [/else]
+        [/if]
+    [/event]
+    
+    #
+    # Ynyr spawns a few more ghosts if there are no more remaining.
+    #
+    [event]
```
Comment (AI0867) [inline data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg]: Won't this event trigger the first time any unit dies and there are no ghosts on the map? So also before the adepts are spawned?

File: data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg [hunk 3]
```diff
@@ -606,6 +617,205 @@
         [/message]
     [/event]
 
+    #
+    # These two events creates enemy naga units that were pillaging the ship
```
Comment (AI0867) [inline data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg]: naga units?

Comment (Earth-Cake) [reply data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg]: I initially thought to put naga there, and it looks like I forgot to change it.

File: data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg (hunk 1 above)
Comment (Earth-Cake) [reply data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg]: Sure thing.

File: data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg (hunk 2 above)
Comment (Earth-Cake) [reply data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg]: Well I have to check this but I believe it won't trigger if Ynyr is not present on the map.

**Comment by ln-zookeeper** [comment]
I guess I wouldn't call this an easter egg, or at least I recognize no jokes or references if there are some. There are two dark adept sweethearts at the second shipwreck (what's the purpose of the first?) creating ghosts from the corpses, in a scenario where you're already fighting enemies who are, for all intents and purposes, undead. Certainly, it's a few more enemies you can fight, but it's also a tomato surprise of at least 4 units suddenly spawning around what's probably a lone scout. For a long time we specifically tried to **decrease** the amount of unpredictable fatal surprises like that.

In terms of narrative, what does the first shipwreck have to do with the second one? Did the adepts arrive on one of them? Did they already plunder the first one? There are two shipwrecks that are obviously intended to be connected somehow, but I can't see what the connection is supposed to be. Also, a few scenarios later Baldras will say he's "heard of" necromancers, which wouldn't make much sense if in this scenario he's already sentencing them to death. Sure, even currently Kestrel recruits Dark Adepts that you fight meaning Baldras has more than heard of them, but there's no other interaction with them so it's probably pretty easy for the player to forget.

**Comment by Earth-Cake** [comment]
> I guess I wouldn't call this an easter egg, or at least I recognize no jokes or references (which easter eggs typically involve) if there are some.

I've planned to add a reference about this, and my thoughts were:
a) their life story
b) or some references to the undead (kinda how necromancers make ghosts (or not if that lore already exists))

> ...but it's also a tomato surprise of at least 4 units suddenly spawning around what's probably a lone scout.

I know. This is a thing I am trying to find a way to avoid, but still I didn't find a solution. Anyway, I _tried_ to drop hints that something foul might be going on. I will try to make it more obvious.

> In terms of narrative, what does the first shipwreck have to do with the second one? Did the adepts arrive on one of them? Did they already plunder the first one? There are two shipwrecks that are obviously intended to be connected somehow, but I can't see what the connection is supposed to be. 

Yes, it is supposed that _some_ necromancers already took the corpses for themselves.

> Also, a few scenarios later Baldras will say he's "heard of" necromancers, which wouldn't make much sense if in this scenario he's already sentencing them to death. Sure, even currently Kestrel recruits Dark Adepts that you fight meaning Baldras has more than heard of them, but there's no other interaction with them so it's probably pretty easy for the player to forget.

I think that it can be solved by changing the dialogue a bit if this (event) happens.

File: data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg (hunk 2 above)
Comment (Earth-Cake) [reply data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg]: It will. I fixed that.

**Comment by GregoryLundberg** [comment]
1)  One subject / One PR is best.

What are the bugs? If they're bugs, move them to a separate PR so they can be fixed even if this one does not get approved.

2) I agree, this does not feel like an easter egg. And I agree, this seems like a rotten tomato.

If the idea is to have some justification/use for the ship wrecks, how about having one have a treasure which might help in some small way with the remainder of the campaign, but is not overly powerful. Have a character in the other shipwreck appear and offer it was a reward if you do something. That would probably be something you'd already do, anyway. So the NPC can be there but only thanks the player, without offering the reward, if you've already done what was needed. In this way, it's a prize for doing things in a certain order.

In HttT, I added a book which granted some abilities to the reader (one unit) which helped in a couple of the remaining scenes. Something like that. I added it as a reward for helping a character who was already present but had no purpose. This added a mini-quest for the scene which only slightly modified the way one would play through it.

**Pull request closed without merging**